### PR TITLE
unix time helpers

### PR DESCRIPTION
**Pull request by dpetrovics**
Get the current UnixTime, and get a DateTime object from a given UnixTime

### CODE REVIEW

File: src/cljx/racehub/util/time.cljx [hunk 1]
```diff
@@ -154,6 +166,16 @@
    (time/interval (time/epoch)
                   (timestamp-to-datetime ts))))
 
+(s/defn unix-time :- UnixTime
+  "Returns the current UnixTime"
+  []
+  (int (/ (coerce/to-long (now)) 1000)))
+
+(s/defn unix-time->datetime
+  "Returns the DateTime object for the given UnixTime (seconds since epoch)."
```
Comment (sritchie) [inline src/cljx/racehub/util/time.cljx]: why don't we just store milliseconds?

**Comment by sritchie** [comment]
Let's just kill this and move to milliseconds. That'll make the JS and clj-time interop easier.

File: src/cljx/racehub/util/time.cljx [hunk 2]
```diff
@@ -37,6 +37,18 @@
      (some (fn [regex] (re-find regex s))
            valid-calendar-formats))))
 
+(def MaxUnixTime
```
Comment (sritchie) [inline src/cljx/racehub/util/time.cljx]: remove

File: src/cljx/racehub/util/time.cljx [hunk 3]
```diff
@@ -302,16 +324,16 @@
...
-  `(with-time* ~t (fn [] ~@body)))
+    [t & body]
+    `(with-time* ~t (fn [] ~@body)))
 
   (defn month-name
-  "Takes in a date object and returns the month name."
-  [date-obj]
-  (format/unparse (format/formatter "MMM")
-                  date-obj)))
+    "Takes in a date object and returns the month name."
```
Comment (sritchie) [inline src/cljx/racehub/util/time.cljx]: looks like you added a bunch of whitespace?

File: src/cljx/racehub/util/time.cljx [hunk 4]
```diff
@@ -322,20 +344,28 @@
...
     [unix-time-secs :- UnixTime]
     (time/to-default-time-zone (js-date-from-unix unix-time-secs)))
 
+  (s/defn calendar-str->unix :- (s/maybe time/UnixTime)
+    "Takes in a mm/dd/yyyy and returns the corresponding unix time."
+    [s :- s/Str]
+    (when-let [res (->> s
+                        (format/parse (format/formatter "MM/dd/yyyy"))
+                        (coerce/to-long))]
+      (/ res 1000)))
```
Comment (sritchie) [inline src/cljx/racehub/util/time.cljx]: whoopsies

File: src/cljx/racehub/util/time.cljx [hunk 5]
```diff
@@ -154,14 +166,24 @@
    (time/interval (time/epoch)
                   (timestamp-to-datetime ts))))
 
+(s/defn unix-time :- UnixTime
```
Comment (sritchie) [inline src/cljx/racehub/util/time.cljx]: add tests to roundtrip these, please

File: src/cljx/racehub/util/time.cljx [hunk 6]
```diff
@@ -37,6 +37,18 @@
      (some (fn [regex] (re-find regex s))
            valid-calendar-formats))))
 
+(def MaxUnixTime
+  "Integer/MAX_VALUE (not defined in javascript)"
+  2147483647)
+
+(s/defn valid-unix-time? :- s/Bool
```
Comment (sritchie) [inline src/cljx/racehub/util/time.cljx]: you don't need this anymore

File: src/cljx/racehub/util/time.cljx (hunk 3 above)
Comment (dpetrovics) [inline src/cljx/racehub/util/time.cljx]: not sure what this is, they all seem to be formatted OK for me locally

File: src/cljx/racehub/util/time.cljx (hunk 5 above)
Comment (dpetrovics) [inline src/cljx/racehub/util/time.cljx]: These two are different- one returns the current unix time (no conversion), the other one is a conversion. So no round-tripping here. Just wrappers over to-long and from-long, from the cljs-time.coerce ns

Comment (sritchie) [inline src/cljx/racehub/util/time.cljx]: `(with-time blah (= (unix-time->datetime (unix-time)) (now))`, something like that

Comment (sritchie) [inline src/cljx/racehub/util/time.cljx]: or however the conversions work

Comment (dpetrovics) [inline src/cljx/racehub/util/time.cljx]: added

**Pull request merged**